dnf-behave-tests/dnf/steps/repo.py

```diff
@@ -372,3 +378,22 @@ def given_system(context, system):
 def given_no_osrelease(context):
     behave.use_fixture(osrelease_fixture, context)
     context.scenario.osrelease.delete()
+
+
+@behave.step("I rewrite solvfile version of \"{path}\" with \"{version}\"")
+def rewrite_solvfile_version(context, path, version):
+    assert len(version) == 3, "Version is expected to be 3 characters long"
+    path = path.format(context=context)
+    ensure_file_exists(path)
+    fp = open(path, "r+b")
+    # Seek 40 bytes from start of file to the beginning userdata section
+    fp.seek(40, 0)
+    userdata_dnf_magic = fp.read(4)
+    assert userdata_dnf_magic == b'\x00dnf'
+
+    userdata_dnf_version = fp.read(4)
+    assert userdata_dnf_version == b'\x001.0'
+
+    fp.write(bytes(version, 'utf-8'))
+    fp.write(b'\x00\x00')
```

Comment:
So here you rewrite the `libsolv_toolversion`, the original value of which is `"\0\0\0\0\01.2"` and IIUC after the rewrite the value is `"0.1\0\01.2"`? I guess this does the job but it'd be nicer to rewrite it in a bit cleaner way, how about making it look e.g. like `"\0\0\0bogus"` after the rewrite? The funky string will be much easier to differentiate in error messages, as opposed to ones and dots and zeros which can get mixed up easily.

Reply:
Sure, I have changed it to `"\0\0\0bogus"`. :slightly_smiling_face: